src/components/LeftSidebar/NewGroupConversation/SetContacts/SetContacts.vue

```diff
@@ -219,6 +246,36 @@ export default {
 		focusInput() {
 			this.setContacts.focus()
 		},
+
+		dialType(event) {
+			switch (event.type) {
+			case 'add': {
+				this.searchText += event.value
+				return
+			}
+			case 'subtract': {
+				this.searchText = this.searchText.slice(0, -1)
+				return
+			}
+			case 'confirm': {
+				this.addParticipantPhone()
+				return
+			}
+			case 'set':
+			default: {
+				this.searchText = event.value
+				this.regionCode = event.code
+			}
+			}
+		},
```

Comment:
The idea of synching the typing with events is good . I would suggest another way to do it though 
* Keep the events confirm and set 
* For adding or substracting events, we can instead form the phone number inside `DialpadPanel` and sync it with `searchText` . 

It will at least reduce the redundancy a bit as the phone number composition is handled internally now .

Reply:
Will try to move the logic to the composable